### PR TITLE
Ul2019

### PR DESCRIPTION
**Pull request by whahmad**
Dear Experts,
EW parameters are modified for ultra legacy as recommended here:
https://github.com/cms-sw/genproductions/tree/master/bin/GenValidation/updated_validation_cards/dy_mlm_mll50/dy01234mlm_inc
please check it and merge it.

### CODE REVIEW

**Comment by sihyunjeon** [comment]
This is an EXO analysis and the analysers wanted to synchronise the settings to central DY samples as much as possible so I asked Wael to check the new run cards.
@whahmad Did you also check the run cards as well?

**Comment by whahmad** [comment]
@sihyunjeon i will check it soon

**Comment by whahmad** [comment]
I just modified run card. please check!

**Comment by whahmad** [comment]
question please, why Higgs was not excluded? In the process card here:
https://github.com/cms-sw/genproductions/blob/master/bin/GenValidation/updated_validation_cards/dy_mlm_mll50/dy01234mlm_inc/dyellell01234j_5f_LO_MLM_proc_card.dat

**Comment by agrohsje** [comment]
the model has 0 b mass and hence vanishing yukawa to quarks except top.

**Comment by agrohsje** [comment]
I think I asked this question in the original pr: model_v4 lfv has 0 b mass as well and all yukawa's except top are 0?

File: python/ThirteenTeV/LFV_Z_hadronizer_RunIISummer19UL17.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+externalLHEProducer = cms.EDProducer("ExternalLHEProducer",
+    args = cms.vstring('/afs/cern.ch/user/w/whajahma/public/UL19_LFV_ZtoL1L2_EWsynch/LFV_ZtoL+L-_slc7_amd64_gcc700_CMSSW_10_6_0_tarball.tar.xz'),
+    nEvents = cms.untracked.uint32(5000),
+    numberOfParameters = cms.uint32(1),
+    outputFile = cms.string('cmsgrid_final.lhe'),
+    scriptName = cms.FileInPath('GeneratorInterface/LHEInterface/data/run_generic_tarball_cvmfs.sh')
+)
+
+#Link to datacards:
```
Comment (agrohsje) [inline python/ThirteenTeV/LFV_Z_hadronizer_RunIISummer19UL17.py:11]: can you adjust this to include the proper link.

**Comment by agrohsje** [comment]
Hi @sihyunjeon . Modulo the small correction I am happy to merge. Let me know.

File: python/ThirteenTeV/LFV_Z_hadronizer_RunIISummer19UL17.py (hunk 1 above)
Comment (whahmad) [reply python/ThirteenTeV/LFV_Z_hadronizer_RunIISummer19UL17.py:11]: What do you mean with proper link?

**Comment by agrohsje** [comment]
Just a link to the cards used for the gridpack: 
https://github.com/cms-sw/genproductions/tree/UL2019/bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/LFV_ZtoL%2BL-
The current link points to DY.

**Comment by whahmad** [comment]
Link to datacards has been adjusted.

**Comment by agrohsje** [comment]
+1 from me. @sihyunjeon please let me know if i can merge.

**Comment by sihyunjeon** [comment]
@agrohsje Do we need to merge the pythia cards as well? I guess @whahmad got this card from [2] used for DYJets MLM in the past. The only difference between the pythia card already in genproductions [1] and this one asked for pull request is the TimeShower:mMaxGamma = 4.0 setting. And I don't think this makes a sensitive difference (default value is 10.0).
So I don't think this pythia card has a need to be added to genproudctions as this is a copy paste of CP5 plus slight modification of mMaxGamma setting.

For the UL default showering, will we be using 4.0 or 10.0 for mMaxGamma?

[1] https://github.com/cms-sw/genproductions/blob/master/genfragments/ThirteenTeV/Hadronizer/Hadronizer_TuneCP5_13TeV_MLM_5f_max4j_qCut19_LHE_pythia8_cff.py
[2] https://cms-pdmv.cern.ch/mcm/public/restapi/requests/get_fragment/MUO-RunIIFall18wmLHEGS-00003/0

**Comment by agrohsje** [comment]
Hi @sihyunjeon . I prefer just storing gridpack cards and no fragments. It is best to get them from MCM. Sometimes people like us to review, but strictly speaking having them here is of little use. As we need to clean-up anyway, we can also add for now. I hope once Andreas finished his work, things will be cleaner.
Not sure why we have sometimes 4 sometimes 1 GeV. Will be check if there is a reason.

**Comment by sihyunjeon** [comment]
@agrohsje Yeah I think there is no harm adding this for now. But it would be also nice to know the reason why mMaxGamma are sometimes set differently.

**Comment by agrohsje** [comment]
maxgamma = 4 gev comes from the production in powheg for wz, zz which starts with 4 gev. for ttbar it is 1 as we use mll=1 gev in ttz samples. so this seems indeed consistent and tailored to what is used in powheg and what we specify in mg5.

**Pull request merged**